common/stdout.h

```diff
@@ -23,12 +25,38 @@ class ConsoleHelper : public Print {
       stdout_output != default_output;
   }
   size_t write(uint8_t b) override {
+#ifdef TEENSYDUINO
+#if defined(RAWHID_INTERFACE)	
+	size_t ret = 1;
+    if (stdout_output) {
+		ret = stdout_output->write(b);
+	} else {
+		SerialHID.write(b);   
```

Comment:
Why can't we just make SerialHID extend Print?
Alternatively; why can't we just wrap SerialHID in something that extends Print?

Reply:
That would be the preferred way of implementing it.
I tried to inherit both From Print() and Stream() but did not get it to work (my programming skills failed me to understand why it did not work)
The _usb_rawhid.h inherits from the Teensuiduino usb_rawhid_class RawHID, I tried to extend and managed to get it working the way it is implemented in this PR. The _usb_rawhid.h is my attempt to make the RawHID objetc comply with Stream() to be able to use it in PorffieOS as a serial output, hence my new object _usb_rawusb_class SerialHID

This works;
`class rawHIDAdapter {
public:
  static void begin() { /* nothing todo */ }
  // static bool Connected() { return !!WebUSBSerial; }
  static bool Connected() { return true; }
  static bool AlwaysConnected() { return true; }
  static Stream& stream() { return SerialHID; }
  static const char* response_header() { return "-+=BEGIN_OUTPUT=+-\n"; }
  static const char* response_footer() { return "-+=END_OUTPUT=+-\n"; }
};`

When I try to use it the way it should;
     stdout_output = &SA::stream();
or;
     stdout_output = &SerialHID; //reroute output to SerialHID
I get the following error;
   ...\ProffieOS.ino.cpp.o: In function `RawHIDParser::Loop()':
   ...\ProffieOS/ProffieOS.ino:1572: undefined reference to `SerialHID'
collect2.exe: error: ld returned 1 exit status

I am sure that this could be implemented different to make the SerialHID behave as a Stream* or Print* Although receiving data might prove difficult to get right, and still requires its own parser? 

So far I am quite happy that I got it to work, but realise that it is not worthy integrating into the main OS the way it is written right now. I would need some help to get this fixed, as my coding-skills are not enough to see why this is not working the way it should...

Reply:
There are a few options for how to make this work....

1. Start with a copy of usb_seremu.{h,c} from the teensy library, then just modify that to talk to the rawhid endpoint. This would make it work exactly like a serial port, so it would be compatible with everything.
2. Don't copy any code, just write a new piece of code in ProffieOS that extends Stream and calls the rawhid recv/send functions to send/receive data. This new class could inherit from Looper to do the polling required. I think this might be slightly more complicated, because there may be some buffere management involved.